File: tests/matlab_octave/testLaplacian.m
classdef testLaplacian < matlab.unittest.TestCase
    methods(Test)
        function testNullityofLaplacian(testCase)
            addpath ('../../src/matlab_octave')
            
            ks=[2,4,6,8];
            tol = 1e-10;
            for k = ks
            	m = 2*k+1;
            	dx=1/m;
            	
            	L= lap(k,m,dx);
            	field = ones(m+2,1);
            	sol = L * field;
            	
            	testCase.verifyLessThan(norm(sol),tol, ...
            		sprintf("Nullity test failed for k = %d", k));
             end
        end
    end
end

File: julia/MOLE.jl/test/testLaplacian.jl
using Test, LinearAlgebra

include("../src/MOLE.jl")
using .MOLE

tol = 1e-10

@testset "Testing Laplacian for order k=$k" for k=2:2:8
    m = 2*k+1
    L = MOLE.lap(k, m, 1/m)
    field = ones(m+2, 1)
    sol = L*field
    @test norm(sol) < tol
end;